jwst/srctype/srctype.py

```diff
@@ -67,8 +73,30 @@ def set_source_type(input_model):
 
             log.info('Input SRCTYPE is unknown; setting default SRCTYPE = %s' % src_type)
 
+        # Set the source type in the global meta attribute
         input_model.meta.target.source_type = src_type
 
+        # If the input contains one or more slit instances,
+        # set the value in each slit too
+        if isinstance(input_model, datamodels.SlitModel):
+            input_model.source_type = src_type
+
+        elif input_model.meta.exposure.type == 'NRS_FIXEDSLIT':
+
+            # NIRSpec fixed-slit is a special case: Apply the source type
+            # determined above to only the primary slit (the one in which
+            # the target is located). Set all other slits to the default
+            # value, which for NRS_FIXEDSLIT is 'POINT'.
+            default_type = 'POINT'
+            primary_slit = input_model.meta.instrument.fixed_slit
+            log.debug(' primary_slit = {}'.format(primary_slit))
+            for slit in input_model.slits:
+                if slit.name == primary_slit:
+                    slit.source_type = src_type
+                else:
+                    slit.source_type = default_type
+                log.debug(' slit {} = {}'.format(slit.name,slit.source_type))
```

Comment:
You might need a space after the comma.

Reply:
I don't see any flake8 complaints, but I'll fix it anyway, just for readability.